### ISSUE TITLE
Open Development Branch

### ISSUE THREAD

**Issue by egecans** [bounswe/bounswe2023group2#354]
### Issue Description

According to our experience last week with the mobile team, we need a development branch for working separately. Because main-branch pushes are important, we couldn't push the improvements before they are ready. However, some kinds of improvements take days, and until we push them into the main branch it can cause a conflict. Thus, we need a development branch that we can easily push our improvements. When we begin a new branch, we should base this dev branch and continue with our new branch from there. With the usage of the development branch; we can follow what other team-mates do and we may prevent some possible conflicts.

### Deadline of the Issue

18.10.2023

### Reviewer

Merve Gürbüz

**Comment by mehmetkuzu** [comment]
I think it's ok. 

This is a need for all of us. 

One thing: We should make pull requests at possible points to share the current dev by the team to other teams. (Stable ones of course.)

The details can be decided with the relevant team. Will the subteam work on the same branch together? Will they need seperate branches for members?

As the things get complicated the possibility of problems increase for sure.

**Comment by egecans** [comment]
> I think it's ok.
> 
> This is a need for all of us.
> 
> One thing: We should make pull requests at possible points to share the current dev by the team to other teams. (Stable ones of course.)
> 
> The details can be decided with the relevant team. Will the subteam work on the same branch together? Will they need seperate branches for members?
> 
> As the things get complicated the possibility of problems increase for sure.

I think one development branch is enough for all teams with proper usage.
I don't think anyone from the backend team will interfere with the mobile branch. The same goes for other team branches. That's why I think there is no need to open development branches team by team.

**Comment by egecans** [comment]
I created the development branch from the main branch. Then I uploaded my branch into the development on #361
In this way, before my branch is fully completed, I can upload the current (working) branch into the development without waiting any review from someone. 
If everyone uses the development branch, there won't be conflicts while merging branches into the main.
2 things to do with development branch:
 - When you start a new branch you should create that branch from the development branch.
 - You should fetch the lastly updates into your local branch from development branch before you push your branch and want a pull request into the main.

**Comment by mehmetkuzu** [comment]
> I created the development branch from the main branch. Then I uploaded my branch into the development on #361 In this way, before my branch is fully completed, I can upload the current (working) branch into the development without waiting for any review from someone. If everyone uses the development branch, there won't be conflicts while merging branches into the main. Two things to do with the development branch:
> 
> * When you start a new branch you should create that branch from the development branch.
> * You should fetch the last updates into your local branch from the development branch before you push your branch and want a pull request into the main.
> 
> In this way, we will prevent conflicts when merging 3 separate branches into the main for the same team. Example for Mobile Team: Hasan created the network manager, Egecan created the nav structure, and Cahid created authorization pages in their local branches. Then, they open the pull requests on Tuesday morning into the main branch Once any of the pull request is accepted, the remaining ones will give a conflict. This happens because they don't know what the other team members do in the project. But if they used the development branch like the explanation before they created pull requests, there wouldn't be any conflicts.

Thanks. I see now.
That is good to me.

Just to check: Main branch will always reflect a kind of "stable" and checked version.
Development branch is the "main" development branch. And anyone developping own part will branch from dev branch and pull regularly so that has the latest changes in dev branch. And will commit to dev branch. (This does not require PR right? But it may spoil other developpers work?)

**Comment by egecans** [comment]
I closed that issue because it would be the same if it requires a review for pushing this branch. Else, if everyone pushes their PR's at the same time, It would be a conflict again.
The problem was that when everyone in the team opened a Pull Request at the same time, after the first PR was merged to the main branch, the remaining PRs were conflicting. As each PR was taken to the main, the remaining PRs had to resolve these conflicts in their own branches. We decided not to send PRs asynchronously. 
The solution is not opening PRs asynchronously. So there is no need to use dev branch for this issue.
